packages/@aws-cdk/aws-s3/test/auto-delete-objects.test.ts

```diff
@@ -0,0 +1,250 @@
+import '@aws-cdk/assert/jest';
+import * as cdk from '@aws-cdk/core';
+import * as s3 from '../lib';
+import { handler } from '../lib/auto-delete-objects-handler/index';
+
+test('when autoDeleteObjects is enabled, a custom resource is provisioned + a lambda handler for it', () => {
+  const stack = new cdk.Stack();
+
+  new s3.Bucket(stack, 'MyBucket', {
+    removalPolicy: cdk.RemovalPolicy.DESTROY,
+    autoDeleteObjects: true,
+  });
+
+  expect(stack).toHaveResource('AWS::S3::Bucket');
+  expect(stack).toHaveResource('AWS::Lambda::Function');
+  expect(stack).toHaveResource('Custom::AutoDeleteObjects',
+    {
+      BucketName: { Ref: 'MyBucketF68F3FF0' },
+      ServiceToken: {
+        'Fn::GetAtt': [
+          'CustomAutoDeleteObjectsCustomResourceProviderHandlerE060A45A',
+          'Arn',
+        ],
+      },
+    });
```

Comment:
Nitpick: We follow the `GIVEN`, `WHEN`, `THEN` naming

 ```suggestion
  // GIVEN
  const stack = new cdk.Stack();

  // WHEN
  new s3.Bucket(stack, 'MyBucket', {
    removalPolicy: cdk.RemovalPolicy.DESTROY,
    autoDeleteObjects: true,
  });

  // THEN
  expect(stack).toHaveResource('AWS::S3::Bucket');
  expect(stack).toHaveResource('AWS::Lambda::Function');
  expect(stack).toHaveResource('Custom::AutoDeleteObjects',
    {
      BucketName: { Ref: 'MyBucketF68F3FF0' },
      ServiceToken: {
        'Fn::GetAtt': [
          'CustomAutoDeleteObjectsCustomResourceProviderHandlerE060A45A',
          'Arn',
        ],
      },
    });
```